src/index-widget/index-widget.ts

```diff
@@ -5,22 +5,24 @@ import {
   forwardRef,
   Optional,
   SkipSelf,
+  OnInit,
+  OnDestroy,
 } from '@angular/core';
-import { Widget, BaseWidget } from '../base-widget';
+import { Widget } from 'instantsearch.js/es/types';
 import { NgAisInstantSearch } from '../instantsearch/instantsearch';
 import indexWidget, {
   IndexWidget,
+  IndexWidgetParams,
 } from 'instantsearch.js/es/widgets/index/index';
-
-const connectIndex = () => indexWidget;
+import { isPlatformBrowser } from '@angular/common';
 
 @Component({
   selector: 'ais-index',
   template: `<ng-content></ng-content>`,
 })
-export class NgAisIndex extends BaseWidget {
-  @Input() public indexName: string;
-  @Input() public indexId?: string;
+export class NgAisIndex implements OnInit, OnDestroy {
```

Comment:
ok with this change, we have almost the same as a widget, except no updateState, which is impossible to create for index, as it doesn't render in the same was as others (the connector is fake)